.github/workflows/rust.yml

```diff
@@ -170,30 +165,35 @@ jobs:
       - name: Ensure that everything builds
         run: |
           eval $(opam env)
-          cargo build --release --all-targets --all-features
+          make all
 
       #
       # Tests
       #
 
-      # https://nexte.st/book/pre-built-binaries.html#using-nextest-in-github-actions
-      - name: Install nextest 0.9.67
+      - name: Install test dependencies
         run: |
-          eval $(opam env)
-          # FIXME: update to 0.9.68 when we get rid of 1.71 and 1.72.
-          cargo install cargo-nextest@=0.9.67 --locked
+          make install-test-deps
 
       - name: Run non-heavy tests
         run: |
           eval $(opam env)
-          cargo nextest run --all-features --release --profile ci -E "not test(heavy)"
+          make nextest-with-coverage
+      - name: Use shared Codecov reporting steps
+        uses: ./.github/actions/codecov-shared
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
 
       - name: Run heavy tests
         if: ${{ matrix.rust_toolchain_version == '1.74' }}
         run: |
           eval $(opam env)
-          cargo nextest run --all-features --release --profile ci -E "test(heavy)"
-
+          make nextest-heavy-with-coverage
```

Comment:
I would run the heavy tests with coverage only on demand.

Reply:
I'll extract it into the Nightly/on-demand.
And yes it is possible it might killed the runner.
![image](https://github.com/user-attachments/assets/d503e84c-11a3-4d03-ab95-231b4cc7df92)

Reply:
Added this to the PR's tasks list.